### ISSUE TITLE
[Extension Bug] writeStore callback isn't a function

### ISSUE THREAD

**Issue by KingDGrizzle** [GilbertGobbels/GAwesomeBot#136]
Seems like the callback is supposed to be optional, but its required..

TypeError: callback is not a function
at serverDocument.save.err (C:\Users\Administrator\Desktop\AuroraBot\Bot\Modules\ExtensionStructures\Extension.js:30:6)
at C:\Users\Administrator\Desktop\AuroraBot\Bot\node_modules\mongoose\lib\model.js:3709:16
at C:\Users\Administrator\Desktop\AuroraBot\Bot\node_modules\mongoose\lib\services\model\applyHooks.js:146:20
at _combinedTickCallback (internal/process/next_tick.js:67:7)
at process._tickDomainCallback [as _tickCallback] (internal/process/next_tick.js:122:9)

**Comment by KingDGrizzle** [comment]
Other bugs with extensions include: you cannot check if the store doesn't exist at it'll crash
Example:
```javascript
if(!extension.store[hi]) channel.createMessage("No hi");
```

**Comment by TTtie** [comment]
🤔 I guess this can be fixable, will wait and make a new pr after #172 

>Other bugs with extensions include: you cannot check if the store doesn't exist at it'll crash
Example:
```js
if(!extension.store["hi"]) channel.createMessage("No hi");```

I guess this is caused by `extension.store` being undefined and is fixable on the extension creator's side. Something like ```js
if (extension.store) {
    if(!extension.store["hi"]) channel.createMessage("No hi");
}```
exists.

**Comment by KingDGrizzle** [comment]
Actually, this isn't a bug-per see. The callback is NOT optional, as it can return the error. This seems invalid in my eyes

**Comment by TTtie** [comment]
Same

**Comment by TTtie** [comment]
I have an idea of prethrowing the error if no callback

**Comment by vladfrangu** [comment]
I mark this issue / pull request as closed. If you have anything against this, ping me.

_This is an semi-automated reply using the Saved Replies function_

## Why did I take this decision?
Mostly invalid, and not an issue. API will change anyways, so its fine
